fn_cve_search/fn_cve_search/components/function_cve_browse.py

```diff
@@ -0,0 +1,118 @@
+        self.options = opts.get("fn_cve_search", {})
+
+        # Getting CVE DataBase URL from app.config file
+        self.CVE_BASE_URL = self.options.get('cve_base_url')+'/{}'
+
+    def _browse_cve_api(self, vendor_name=None):
+        """
+        :param vendor_name: Vendor Name to search for Vulnerabilities from CVE DB
+        :return: if vendor_name=None  then returns JSON with all the vendors, otherwise JSON with all the products
+        associated to a vendor
+        """
+
+        if vendor_name:
+            browse_api = '{}/{}'.format(self.CVE_BASE_URL.format('browse'), vendor_name)
+        else:
+            browse_api = self.CVE_BASE_URL.format('browse')
+        return make_rest_api_get_call(browse_api, api_call='browse')
+
+    def _cve_db_information(self):
+        """
+        :return:more information about the current databases in use and when it was updated
+        """
+        db_info_url = '{}'.format(self.CVE_BASE_URL.format('dbInfo'))
+        return make_rest_api_get_call(db_info_url, api_call='db')
+
+    def _parse_browse_results(self, api_data):
+        """
+        :param api_data: Rest API Call Result Data to Parsed.
+        :return: returning an array  of product and vendor data
+        """
+        tmp_browse_list = []
+        if isinstance(api_data, list):
+            raise NotImplementedError()
+        elif isinstance(api_data, dict):
+            for key_data, value_data in api_data.items():
+                tmp_browse_list.append({key_data: value_data})
+            return tmp_browse_list
+        else:
+            raise NotImplementedError()
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_cve_search", {})
+
+    @function("function_cve_browse")
+    def _function_cve_browse_function(self, event, *args, **kwargs):
+        """Function: A Function to Browse Common Vulnerability Exposures  Vendors and Product & Database information from https://cve.circl.lu Data Base."""
+        try:
+            # Get the function parameters:
+            cve_browse_data = kwargs.get("cve_browse_data")  # text
+            if cve_browse_data:
+                cve_browse_data = cve_browse_data.strip()
+
+            cve_browse_criteria = kwargs.get("cve_browse_criteria")  # text
+            if cve_browse_criteria:
+                cve_browse_criteria = cve_browse_criteria.strip()
+
+            cve_vendor = kwargs.get("cve_vendor")  # text
+            if cve_vendor:
+                cve_vendor = cve_vendor.strip()
+
+            # Variables to Store Parsed CVE API Data
+            result_data_dict = dict()
+
+            log = logging.getLogger(__name__)
+            log.info("cve_browse_data: %s", cve_browse_data)
+            log.info("cve_browse_criteria: %s", cve_browse_criteria)
+            log.info("cve_vendor: %s", cve_vendor)
+
+            yield StatusMessage("starting...")
```

Comment:
Something like: "Getting list of Vendors from the CVE Database"

Reply:
ok